src/bm_sim/CMakeLists.txt

```diff
@@ -0,0 +1,96 @@
+# Include directories
+include_directories(
+  ${CMAKE_CURRENT_SOURCE_DIR}/../BMI
+  ${CMAKE_CURRENT_SOURCE_DIR}/../bf_lpm_trie
+)
+
+# Configure version.cpp
+configure_file(
+  ${CMAKE_CURRENT_SOURCE_DIR}/version.cpp.in
+  ${CMAKE_CURRENT_BINARY_DIR}/version.cpp
+  @ONLY
+)
+
+# Create bmsim library
+add_library(bmsim OBJECT
+  _assert.cpp
+  action_profile.cpp
+  actions.cpp
+  ageing.cpp
+  bytecontainer.cpp
+  calculations.cpp
+  checksums.cpp
+  conditionals.cpp
+  context.cpp
+  control_action.cpp
+  counters.cpp
+  crc_map.cpp
+  debugger.cpp
+  deparser.cpp
+  dev_mgr.cpp
+  dev_mgr_bmi.cpp
+  dev_mgr_packet_in.cpp
+  enums.cpp
+  event_logger.cpp
+  expressions.cpp
+  extern.cpp
+  fields.cpp
+  headers.cpp
+  header_unions.cpp
+  learning.cpp
+  lookup_structures.cpp
+  logger.cpp
+  match_error_codes.cpp
+  match_units.cpp
+  match_tables.cpp
+  md5.c
+  meters.cpp
+  options_parse.cpp
+  P4Objects.cpp
+  packet.cpp
+  parser.cpp
+  parser_error.cpp
+  pcap_file.cpp
+  pipeline.cpp
+  periodic_task.cpp
+  port_monitor.cpp
+  phv.cpp
+  phv_source.cpp
+  stateful.cpp
+  switch.cpp
+  simple_pre.cpp
+  simple_pre_lag.cpp
+  source_info.cpp
+  stacks.cpp
+  tables.cpp
+  table_apply.cpp
+  target_parser.cpp
+  transport.cpp
+  transport_nn.cpp
+  ${CMAKE_CURRENT_BINARY_DIR}/version.cpp
```

Comment:
Please move this below the list of files from the current dir (e.g., move after xxhash.c) to avoid breaking up the list of files from the current dir. I know it is generated from a file in the current dir.